File: user-service/src/main/java/com/gallery/userservice/controller/NotificationController.java
package com.gallery.userservice.controller;

import com.gallery.userservice.service.notification.NotificationService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/notify")
public class NotificationController {

    private final NotificationService notificationService;

    public NotificationController(NotificationService notificationService) {
        this.notificationService = notificationService;
    }

    @PostMapping("/email")
    public ResponseEntity<Void> notifyByEmail(@RequestParam String to, @RequestParam String message) {
        notificationService.notifyByEmail(to, message);
        return ResponseEntity.ok().build();
    }

    @PostMapping("/sms")
    public ResponseEntity<Void> notifyBySms(@RequestParam String to, @RequestParam String message) {
        notificationService.notifyBySms(to, message);
        return ResponseEntity.ok().build();
    }
}

File: app/UserService/Controllers/NotificationController.cs
using Microsoft.AspNetCore.Mvc;
using UserService.Services.Notifications;

namespace UserService.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class NotificationController : ControllerBase
    {
        private readonly NotificationService _notificationService;

        public NotificationController(NotificationService notificationService)
        {
            _notificationService = notificationService;
        }

        [HttpPost("email")]
        public ActionResult SendEmail([FromQuery] string email, [FromQuery] string message)
        {
            if (string.IsNullOrWhiteSpace(email) || string.IsNullOrWhiteSpace(message))
                return BadRequest("Email and message are required.");

            bool result = _notificationService.NotifyByEmail(email, message);
            return result ? Ok("Email sent.") : StatusCode(500, "Failed to send email.");
        }

        [HttpPost("sms")]
        public ActionResult SendSMS([FromQuery] string phone, [FromQuery] string message)
        {
            if (string.IsNullOrWhiteSpace(phone) || string.IsNullOrWhiteSpace(message))
                return BadRequest("Phone and message are required.");

            bool result = _notificationService.NotifyBySms(phone, message);
            return result ? Ok("SMS sent.") : StatusCode(500, "Failed to send SMS.");
        }


    }
}
